### PR TITLE
Fix for segment/count Metric Not Emitting with Statsd-emitter

### PR DESCRIPTION
**Pull request by YongGang**
### Description

This pull request resolves a bug where the `segment/count` metric was not being emitted as expected when using the Statsd-emitter. The issue was traced to a discrepancy in the `serviceName` used within the metric name. While the Statsd-emitter was correctly appending the serviceName to the metric name, the configured metric name within the Statsd-emitter did not include the correct serviceName. This mismatch led to a failure during the metric name comparison check, resulting in the segment/count metric not being reported. The configuration has been corrected to include the appropriate serviceName, ensuring that the segment/count metric is now emitted correctly for each service.

#### Release note

<hr>

This PR has:

- [x] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

**Review by TSFenwick — APPROVED** [review]
Lgtm

**Review by suneet-s — COMMENTED** [review]
Good find!

Can you update the docs for statsd and prometheus as well to indicate the correct service name.

Also, it looks like some service names are configurable - like the coordinator, overlord and broker, but historicals, peons, middle managers are hardcoded. Can the docs make reference to these configurations so that users know how to decide what service name to use.

**Comment by YongGang** [comment]
> Good find!
> 
> Can you update the docs for statsd and prometheus as well to indicate the correct service name.
> 
> Also, it looks like some service names are configurable - like the coordinator, overlord and broker, but historicals, peons, middle managers are hardcoded. Can the docs make reference to these configurations so that users know how to decide what service name to use.

Doc Updated

File: docs/development/extensions-contrib/prometheus.md [hunk 1]
```diff
@@ -107,8 +107,10 @@ For metrics which are emitted from multiple services with different dimensions,
 the service name. For example:
 
 ```json
-"coordinator-segment/count" : { "dimensions" : ["dataSource"], "type" : "gauge" },
-"historical-segment/count" : { "dimensions" : ["dataSource", "tier", "priority"], "type" : "gauge" }
+"druid/coordinator-segment/count" : { "dimensions" : ["dataSource"], "type" : "gauge" },
+"druid/historical-segment/count" : { "dimensions" : ["dataSource", "tier", "priority"], "type" : "gauge" }
 ```
+
+Please note that the Coordinator service name can be customized via the `druid.selectors.coordinator.serviceName` configuration setting. It is essential to ensure that this name matches the service name prefix used in the metric configuration.
```
Comment (suneet-s) [inline docs/development/extensions-contrib/prometheus.md:114]:
```suggestion
```

File: docs/development/extensions-contrib/statsd.md [hunk 2]
```diff
@@ -65,7 +65,9 @@ e.g.
 For metrics which are emitted from multiple services with different dimensions, the metric name is prefixed with
 the service name.
 e.g.
-`"coordinator-segment/count" : { "dimensions" : ["dataSource"], "type" : "gauge" },
- "historical-segment/count" : { "dimensions" : ["dataSource", "tier", "priority"], "type" : "gauge" }`
- 
+`"druid/coordinator-segment/count" : { "dimensions" : ["dataSource"], "type" : "gauge" },
+ "druid/historical-segment/count" : { "dimensions" : ["dataSource", "tier", "priority"], "type" : "gauge" }`
+
+Please note that the Coordinator service name can be customized via the `druid.selectors.coordinator.serviceName` configuration setting. It is essential to ensure that this name matches the service name prefix used in the metric configuration.
```
Comment (suneet-s) [inline docs/development/extensions-contrib/statsd.md:71]:
```suggestion
```

**Review by suneet-s — APPROVED** [review]
I think I mis spoke earlier. It looks like the service name for the coordinator is hard coded in CliCoordinator. Thanks for the fix!

```
binder.bindConstant()
                  .annotatedWith(Names.named("serviceName"))
                  .to(TieredBrokerConfig.DEFAULT_COORDINATOR_SERVICE_NAME);
```

**Pull request merged**